docs/usage/user.asciidoc

```diff
@@ -0,0 +1,396 @@
+<4> How to use all possible user fields is detailed below.
+
+[discrete]
+[[ecs-user-usage-field-reuse]]
+===== Field reuse
+
+The user fields can be reused (or appear) in many places across ECS. This makes
+it possible to capture many users relevant to a single event.
+
+Here's the full list of places where the user fields can appear:
+
+* `user.*`
+* `user.effective.*`
+* `user.target.*`
+* `user.changes.*`
+* `source.user.*`
+* `destination.user.*`
+* `client.user.*`
+* `server.user.*`
+* `host.user.*` (<<ecs-user-usage-deprecations,deprecated>>)
+
+Let's go over the meaning of each.
+
+The examples below will only populate `user.name` and sometimes `user.id` inside
+the various `user` nestings, for readability.
+However in implementations, unless otherwise noted, all `user` fields that can
+reasonably be populated in each location should be populated.
+
+[discrete]
+[[ecs-user-usage-user-at-root]]
+====== User fields at the Root of an Event
+
+The user fields at the root of an event are used to capture the user
+performing the main action described by the event. This is especially important
+when there's more than one user present on the event. `user.*` fields at the root
+of the event represent the user performing the action.
+
+In many cases, events that only mention one user should populate the user fields
+at the root of the event, even if the user is not the one performing the action.
+
+In cases where a purpose-specific user field such as `url.username` is populated,
+`user.name` should also be populated with the same user name.
+
+[source,json]
+-----------
+{
+  "url": { "username": "alice" }, <1>
+  "user": { "name": "alice" }, <2>
+  "related": { "user": ["alice"] }
+}
+-----------
+<1> Purpose-specific username field
+<2> Username copied to `user.name` to establish `user.name` as a reliable baseline.
+
+[discrete]
+[[ecs-user-usage-remote-logons]]
+====== Remote Logons
+
+When users are crossing host boundaries, the users are captured at
+`source.user` and `destination.user`.
+
+Examples of data sources where this is applicable:
+
+* Remote logons via ssh, kerberos
+* Firewalls observing network traffic
+
+In order to align with ECS' design of having `user` at the root of the event as the
+user performing the action, all `source.user` fields should be copied to `user` at the root.
+
+Here's an example where user "alice" logs on to another host as user "deus":
+
+[source,json]
+-----------
+{
+  "user": {
+    "name": "alice"
+  },
+  "source": {
+    "user": {
+      "name": "alice"
+    },
+    "ip": "10.42.42.42"
+  },
+  "destination": {
+    "user": {
+      "name": "deus"
+    },
+    "ip": "10.42.42.43"
+  },
+  "related": { "user": ["alice", "deus"] }
+}
+-----------
+
+Whenever an event source populates the `client` and `server` fields in addition
```

Comment:
I could see a second example which builds on the first, showing both `client`/`server` + `source`/`destination` fields all populated in the single event. WDYT?

Reply:
Here I think I would err on the side of "terseness", if we can call such a long page terse. If the idea is to reinforce this pattern, perhaps we can instead link to the page on mapping network events?

Reply:
I've added a link at the end of this paragraph